code/datums/components/twohand.dm

```diff
@@ -0,0 +1,92 @@
+/obj/item/proc/is_twohandable() // Returns TRUE if an object has a two-handing component. FALSE otherwise.
+	if(GetComponent(/datum/component/twohand))
+		return TRUE
+
+/obj/item/proc/is_twohanded() // Returns TRUE if an item is wielded with two hands. FALSE otherwise.
+	/datum/component/twohand/T = GetComponent(/datum/component/twohand)
+	if(istype(T))
+		return T.wielded
+
+
+/*
+ * Component for twohanding an /obj/item.
+ * Default behavior is to toggle wielding with attack_self, however this can be overridden by changing the toggle_wield_signal in the Initialize call.
+ */
+/datum/component/twohand
+	var/wielded
+	var/obj/item/container
```

Comment:
you'll want to disassociate this from the object when the component gets destroyed.

Either that or use soft references with `UID()`